### DISCUSSION TITLE
Watchdog

### DISCUSSION THREAD

**Discussion by bluesky81** [JAndrassy/ArduinoOTA#81] [General]
Hello,
thanls for the great work.
Is it possible to use a watchdog und deactivate it since update is active?

Greetings!

**Comment by JAndrassy** [comment]
so this one I must improve. 
there is `beforeApply` callback, but that is called only after the upload and before applying the update. So this works for long watchdog timeout and having `ArduinoOTA.poll();` as first after wdt reset.

in setup()
```
ArduinoOTA.beforeApply(otaStart);
```
```
void otaStart() {
  // reconfigure watchdog here
}
```
I use ArduinoOTA with watchdog but I have the watchdog set to 16 seconds (2 x 8 sec).
